src/coreclr/src/jit/codegenlinear.cpp

```diff
@@ -2382,7 +2382,8 @@ void CodeGen::genCodeForJumpTrue(GenTreeOp* jtrue)
 
 #if defined(TARGET_XARCH)
     if ((condition.GetCode() == GenCondition::FNEU) &&
-        (relop->gtGetOp1()->GetRegNum() == relop->gtGetOp2()->GetRegNum()))
+        (relop->gtGetOp1()->GetRegNum() == relop->gtGetOp2()->GetRegNum()) &&
+        !relop->gtGetOp1()->isUsedFromSpillTemp() && !relop->gtGetOp2()->isUsedFromSpillTemp())
```

Comment:
We can have the same register assigned to the two operands, even if they are not the same, if one of them is spilled and marked `RegOptional`. This was exposed by a jitStressRegs test run where the available register set is intentionally limited.